tests/neg/LiquidClass1.hs

```diff
@@ -0,0 +1,39 @@
+module LiquidClass where
+
+
+-- | Typing classes
+-- | Step 1: Refine type dictionaries:
+
+class Compare a where
+	cmax :: a -> a -> a
+	cmin :: a -> a -> a
+
+instance Compare Int where	
+{-@ instance Compare Int where 
+	cmax :: Odd -> Odd -> Odd ;
```

Comment:
The reason why I believe we need the ';' is that

If you go to your GHC.Real.Spec and move the type specification of
`fromIntegral` AFTER the Class specification, then the type specification
for `fromIntegral` would be interpreted as if it belonged to the class
methods,
which is not what we want.

On Mon, Jan 12, 2015 at 11:09 AM, Eric Seidel <notifications@github.com>
wrote:

> In tests/neg/LiquidClass1.hs
> <https://github.com/ucsd-progsys/liquidhaskell/pull/304#discussion-diff-22813162>
> :
>
> > @@ -0,0 +1,39 @@
> > +module LiquidClass where
> > +
> > +
> > +-- | Typing classes
> > +-- | Step 1: Refine type dictionaries:
> > +
> > +class Compare a where
> > +	cmax :: a -> a -> a
> > +	cmin :: a -> a -> a
> > +
> > +instance Compare Int where	
> > +{-@ instance Compare Int where
> > +	cmax :: Odd -> Odd -> Odd ;
>
> I'm not sure, I think div and mod are used in a few tests, but I doubt we
> have an instance declared anywhere. Regardless, it parses properly :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/pull/304/files#r22813162>.
>



-- 
Niki Vazou